tests/std/tests/P1502R1_standard_library_header_units/test.cpp

```diff
@@ -804,7 +804,13 @@ int main() {
 
     {
         puts("Testing <syncstream>.");
-        puts("(TRANSITION, not yet implemented.)");
+        syncbuf sync_buf{nullptr};
+        assert(sync_buf.get_wrapped() == nullptr);
+        assert(sync_buf.get_allocator() == std::allocator<char>{});
+        assert(sync_buf.emit() == false);

```

Comment:
This line of code is causing the failure of `P1502R1_standard_library_header_units`. I ran it locally and commenting out this line make it pass. This function call should return `false` with no operations to be executed.

Reply:
Have you tested if it passes without the `assert`? (only the call to emit)

Reply:
On my local machine, the call itself makes the test fails.
```C++
assert(sync_buf.emit() == false); // fails, exit code: 3221225477
assert(sync_buf.emit() == true); // fails. Assertion failed ... line 810
sync_buf.emit(); // fails, exit code: 3221225477
```

Reply:
Follow up:
I found out that this failure happens when using `/MTd` compiler flag. I am still not sure what this means so I'll update the comment if I found anything relevant :cat:

Reply:
I can reduce and report this compiler bug.

Reply:
Reported internally as VSO-1275701 "Standard Library Header Units: Runtime crash with dynamic atexit destructor for syncstreams". I'll see if it continues to repro after the satellite DLL change; if it does, I'll disable the new test code for header units.